#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.507990318388395 -0.29733451543310974 -0.808416985444897 4.025509866896948
  position 1.1666190045063787 -1.4149173455940118 2.0066790997378714
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0.45 -0.45 0
  floorTileSize 0.2 0.2
}
Solid {
  translation 0.2 -0.3 0.05
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  boundingObject USE box
}
Solid {
  translation 0.4 -0.8 0.05
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(1)"
  boundingObject USE box
}
Solid {
  translation 0.8 -0.8 0.05
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(7)"
  boundingObject USE box
}
Solid {
  translation 0.5 -0.6 0.05
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(3)"
  boundingObject USE box
}
Solid {
  translation 0.2 -0.1 0.05
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(4)"
  boundingObject USE box
}
Solid {
  translation 0.7 -0.2 0.05
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(2)"
  boundingObject USE box
}
Solid {
  translation 0.5 -0.3 0.05
  rotation 0 0 1 1.5707996938995747
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(5)"
  boundingObject USE box
}
Solid {
  translation 0.8 -0.5 0.05
  rotation 0 0 1 1.5707996938995747
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(8)"
  boundingObject USE box
}
Solid {
  translation 0.1 -0.4 0.05
  rotation 0 0 1 1.5707996938995747
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(6)"
  boundingObject USE box
}
Solid {
  translation 0.1 -0.7 0.05
  rotation 0 0 1 1.5707996938995747
  children [
    DEF box Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.3 0.1 0.1
      }
    }
  ]
  name "solid(9)"
  boundingObject USE box
}
Robot {
  translation 0.4 -0.4 0.05
  children [
    DistanceSensor {
      translation -0.000971649 -0.0199942 0
      rotation 0 0 1 1.56
      name "ds_north"
    }
    DistanceSensor {
      translation -0.00496617 0.0100049 0
      rotation 0 0 1 -1.56
      name "ds_south"
    }
    DistanceSensor {
      translation -0.024214 0.000657 0
      name "ds_east"
    }
    DistanceSensor {
      translation 0.015786 0.000657 0
      rotation 0 0 1 3.141592653589793
      name "ds_west"
    }
    LED {
      translation 0 0.03 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "north"
    }
    LED {
      translation 0 -0.03 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "south"
    }
    LED {
      translation 0.03 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "east"
    }
    LED {
      translation -0.03 0 0.05
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            metalness 0
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "west"
    }
    DEF BOX Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0901961 0.0235294 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
    }
  ]
  controller "Robot_unit1_Controller"
  supervisor TRUE
}
